lib/board.rb

```diff
@@ -0,0 +1,67 @@
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+
+  end
+
+  def reset!
+    self.cells = [" "] * 9
+  end
+
+  def display
+    d = self.cells
+
+    puts "-----------"
+    puts " #{d[0]} | #{d[1]} | #{d[2]} "
+    puts "-----------"
+    puts " #{d[3]} | #{d[4]} | #{d[5]} "
+    puts "-----------"
+    puts " #{d[6]} | #{d[7]} | #{d[8]} "
+    puts "-----------"
+  end
+
+  def position(location)
+    c = self.cells
+
+    c[location.to_i - 1]
+  end
+
+  def turn_count
+    c = self.cells
+
+    c.select{|e| e.strip != ""}.size
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```